python/smqtk/algorithms/saliency/sbsm.py

```diff
@@ -0,0 +1,208 @@
+import numpy as np
+from tqdm import tqdm
+import copy
+from PIL import Image
+import os
+from sklearn.metrics.pairwise import euclidean_distances
+import six
+from smqtk.algorithms import SmqtkAlgorithm
+from smqtk.algorithms.saliency import SaliencyBlackbox,ImageSaliencyAugmenter
+from smqtk.representation.data_element.memory_element import DataMemoryElement
+
+class SBSM_SaliencyBlackbox (SaliencyBlackbox):
+    
+    """
+    Blackbox function that produces some floating point scalar value for a
+    given descriptor element.
+    :param query_f: Feature of query image
+    :param type: smqtk.representation.DescriptorElement
+    :param base_descr: Base image descriptor
+    :param type: smqtk.representation.DescriptorElement
+        """
+    def __init__(self, query_f, base_descr):
+        
+        self.query_f = query_f
+        
+        self.base_descr = base_descr
+
+    @classmethod
+    def is_usable(cls):
+        """
+        Check whether this implementation is available for use.
+        Required valid presence of query image feature and base image descriptor
+        :return:
+            Boolean determination of whether this implementation is usable.
+        :rtype: bool
+        """
+        
+        return self.base_descr and self.query_f
+        
+    @classmethod
+    def from_iqr_session(cls, iqrs, descr_gen, base_image): 
+        """
+        Create an ``SaliencyBlackbox`` instance from an
+        :class:`smqtk.iqr.IqrSession` instance.
+        Not all implementations of ``SaliencyBlackbox`` implement
+        this method and may raise a ``NotImplementedError`` exception.
+        :raises NotImplementedError:
+            Construction from a ``IqrSession`` is not defined for this
+            implementation.
+        :return: A new instance of of a class implementing the
+            ``SaliencyBlackbox`` class.
+        :rtype: SaliencyBlackbox
+        """
+
+        assert iqrs
+        assert descr_gen.is_usable
```

Comment:
This should not be necessary as the descriptor generator ``is_usable`` class-method is checked at construction, i.e. the instance could not have been constructed if this was false. Also this is a method and this assert merely checks that the method exists.

Reply:
Removed "assert descr_gen.is_usable"